integration_tests/testsuite/test_logging_standard.py

```diff
@@ -52,18 +54,26 @@ def runTest(self):
             self.assertTrue(self._read_log(client, log_name, token, level),
                             'Log entry not found for posted token!')
 
-    @retry(wait_fixed=4000, stop_max_attempt_number=8)
+    @retry(wait_fixed=8000, stop_max_attempt_number=10)
     def _read_log(self, client, log_name, token, level):
+        currentLog = 0
         project_id = test_util._project_id()
         FILTER = 'logName = projects/{0}/logs/' \
-                 '{1}'.format(project_id, log_name)
-        for entry in client.list_entries(filter_=FILTER):
+                 '{1} AND NOT textPayload:"/_ah/health"' \
+                 .format(project_id, log_name)
+        for entry in client.list_entries(filter_=FILTER,
+                                         order_by='timestamp desc'):
+            currentLog += 1
             # since the logs we're examining are for the deployed flex app,
             # we can safely log from the test driver without contaminating
             # the logs under examination.
+            logging.debug(currentLog)
             logging.debug(entry.payload)
             if token in entry.payload:
                 logging.info('Token {0} found in '
                              'Stackdriver logs!'.format(token))
                 return True
+            if currentLog == LOG_PAGE_SIZE:
```

Comment:
Actually, it looks like it might be: https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/list

> The presence of nextPageToken in the response indicates that more results might be available.

Still doesn't really explain why it's timing out though.